engine/resource/src/resource_mounts_file.cpp

```diff
@@ -0,0 +1,219 @@
+
+#include <stdio.h>
+#include <ctype.h>
+
+#include "resource_mounts.h"
+#include "resource_mounts_file.h"
+
+#include <dlib/dstrings.h>
+#include <dlib/log.h>
+
+namespace dmResourceMounts
+{
+const int MOUNTSFILE_HEADER_VERSION = 1;
+const char* MOUNTSFILE_CSV_SEPARATOR = "@,@";
```

Comment:
I didn't want to use a commonly occurring character.
We need to support file paths and http urls, and I don't want to involve any encoding.

Reply:
Example:
```
VERSION@,@1
MOUNT@,@10@,@liveupdate@,@mutable:/Users/mawe/Library/Application Support/57e7d5671c7d94b92e758e6b0ed0f7356a78c957/liveupdate.arci
```